klippa/src/layout.rs

```diff
@@ -512,6 +519,211 @@ impl<'a> Serialize<'a> for ClassDefFormat2<'a> {
     }
 }
 
+impl<'a> SubsetTable<'a> for CoverageTable<'a> {
+    type ArgsForSubset = ();
+    type Output = ();
+    fn subset(
+        &self,
+        plan: &Plan,
+        s: &mut Serializer,
+        args: Self::ArgsForSubset,
+    ) -> Result<(), SerializeErrorFlags> {
+        match self {
+            CoverageTable::Format1(sub) => sub.subset(plan, s, args),
+            CoverageTable::Format2(sub) => sub.subset(plan, s, args),
+        }
+    }
+}
+
+impl<'a> SubsetTable<'a> for CoverageFormat1<'a> {
+    type ArgsForSubset = ();
+    type Output = ();
+    fn subset(
+        &self,
+        plan: &Plan,
+        s: &mut Serializer,
+        _args: Self::ArgsForSubset,
+    ) -> Result<Self::Output, SerializeErrorFlags> {
+        let glyph_count = (self.glyph_count() as usize).min(plan.font_num_glyphs);
+        let Some(glyph_array) = self.glyph_array().get(0..glyph_count) else {
+            return Err(SerializeErrorFlags::SERIALIZE_ERROR_READ_ERROR);
+        };
+
+        let num_bits = 16 - (glyph_count as u16).leading_zeros() as usize;
+        let retained_glyphs: Vec<u32> =
+            if glyph_count > (plan.glyphset_gsub.len() as usize) * num_bits {
```

Comment:
can you add a comment that explains the two different if branches do the same thing and we're just picking which approach will be faster depending on the number of glyphs.